tests/integration/tests/test_clustering.py

```diff
@@ -96,6 +96,36 @@ def test_no_remove(instances: List[harness.Instance]):
     assert len(nodes) == 3, "worker node should not have been removed from cluster"
 
 
+@pytest.mark.node_count(2)
+@pytest.mark.bootstrap_config(
+    (config.MANIFESTS_DIR / "bootstrap-skip-service-stop.yaml").read_text()
+)
+def test_skip_services_stop_on_remove(instances: List[harness.Instance]):
+    cluster_node = instances[0]
+    joining_cp = instances[1]
+
+    join_token = util.get_join_token(cluster_node, joining_cp)
+    util.join_cluster(joining_cp, join_token)
+
+    util.wait_until_k8s_ready(cluster_node, instances)
+    nodes = util.ready_nodes(cluster_node)
+    assert len(nodes) == 2, "nodes should have joined cluster"
+
+    cluster_node.exec(["k8s", "remove-node", joining_cp.id])
+    nodes = util.ready_nodes(cluster_node)
+    assert len(nodes) == 2, "cp node should not have been removed from cluster"
```

Comment:
I don't understand something, why do we expect that the second node is not removed? the annotation is only supposed to stop the services from being stopped, right? does this mean the node won't effectively leave the cluster?

Reply:
That's a fair point. TBH, I have not yet tested this integration tests as the build fails without https://github.com/canonical/k8s-snap-api/pull/8 (and I was too lazy yesterday evening to do a local `replace` in the `go.mod` :) )
Will update the test together with the `k8s-snap-api` version update